### PR TITLE
chore: add dev container

### PR DESCRIPTION
**Pull request by ctran88**
## What's New?

- Adds a dev container config file to bootstrap the dev environment
- Replaces openapitools npm package with docker image to codegen for consistency across sdks

**To test:**
1. In VS Code, download the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
2. If a popup on your lower right window shows up that asks to **"Reopen in Container"**, select it. Skip to step 5
    
    image

3. `cmd + shift + P` to bring up command palette
4. Start typing "rebuild" and select **"Dev Containers: Rebuild and Reopen in Container"**
5. The first time may take a minute or two as it downloads and builds the docker image. Progress can be tracked by clicking the popup

    image

6. Make sure you have your `.env` file populated 
7. Try running linter, formatter, and tests by following commands in the `on-pull-request.yml` CI file or following the documentation for the tools being used
8. Check the extensions and you should find the basics of what you need to dev

**Without VS Code**
If you want to try it without VS Code, you can do so by using the [Dev Container CLI](https://github.com/devcontainers/cli?tab=readme-ov-file#npm-install)

**Codegen change**
codegen change can be tested with
```
curl -o openapi.json https://api.swaggerhub.com/apis/passage/passage-mgmt-sdk-api/1
./generate.sh openapi.json
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have manually tested my code thoroughly
- [ ] I have added/updated inline documentation for public facing interfaces if relevant
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	},
+	"customizations": {
+		"vscode": {
+			"extensions": [
+				"ms-python.python",
+				"charliermarsh.ruff",
+				"arahata.linter-actionlint",
+				"tamasfe.even-better-toml",
+				"-ms-python.autopep8",
+				"-dbaeumer.vscode-eslint"
```
Comment (ctran88) [inline .devcontainer/devcontainer.json:16]: removes the auto-installed extensions from vs code

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			]
+		}
+	},
+	"postCreateCommand": "pip install pytest python-dotenv build ruff"
```
Comment (ctran88) [inline .devcontainer/devcontainer.json:20]: required packages to [test](https://github.com/passageidentity/passage-python/blob/67c0e12ae65adaf0ddb085a3c185a53390a3302c/.github/workflows/on-pull-request.yml#L43) and [build](https://github.com/passageidentity/passage-python/blob/67c0e12ae65adaf0ddb085a3c185a53390a3302c/.github/workflows/deploy.yml#L23) sdk

**Review by ctran88 — COMMENTED** [review]
linting and formatting expected to fail since they will be addressed in a different PR

**Review by vanessa-passage — APPROVED** [review]
Ran dev container

**Comment by vanessa-passage** [comment]
Can we add the info to the contributing file too?

**Comment by ctran88** [comment]
> Can we add the info to the contributing file too?

i added another ticket for it since there might be some additional changes once the e2e tests are pulled out.

**Pull request merged**